### ISSUE TITLE
Bidaf model with lazy parameter for dataset set to True

### ISSUE THREAD

**Issue by Senagb** [allenai/allennlp#1774]
**Describe the bug**
I have ran two Biaf model training on same train and val dataset but with the same model parameter values except that one machine with low GPU memory I used lazy attribute true and max_instances_in_memory: 8 and on the other machine I removed the laziness and cap of max_instances here are the two logs:

With Lazy = true:
"/usr/bin/python3.6 /home/amagdy/Downloads/allennlp-as-a-library-example-master/ProgramticalBidaf.py
{'drop_out': 0.2, 'learning_rate': 0.001920946131952178, 'mini_batch': 64}
<class 'allennlp.data.tokenizers.character_tokenizer.CharacterTokenizer'>
63854it [03:28, 306.84it/s]
<class 'allennlp.modules.text_field_embedders.text_field_embedder.TextFieldEmbedder'>
100%|##########| 610977/610977 [00:23<00:00, 25858.47it/s]
<class 'allennlp.modules.seq2seq_encoders.seq2seq_encoder.Seq2SeqEncoder'>
/usr/local/lib/python3.6/dist-packages/torch/nn/modules/rnn.py:38: UserWarning: dropout option adds dropout after all but last recurrent layer, so non-zero dropout expects num_layers greater than 1, but got dropout=0.2 and num_layers=1
  "num_layers={}".format(dropout, num_layers))
<class 'allennlp.modules.similarity_functions.similarity_function.SimilarityFunction'>
<class 'allennlp.modules.seq2seq_encoders.seq2seq_encoder.Seq2SeqEncoder'>
<class 'allennlp.modules.seq2seq_encoders.seq2seq_encoder.Seq2SeqEncoder'>
start_acc: 0.1837, end_acc: 0.1921, span_acc: 0.1179, em: 0.1238, f1: 0.2037, loss: 6.6961 ||: : 7095it [22:10,  5.33it/s]
start_acc: 0.2362, end_acc: 0.2549, span_acc: 0.1718, em: 0.1780, f1: 0.2667, loss: 5.8957 ||: : 888it [01:05, 13.48it/s]
start_acc: 0.2029, end_acc: 0.2162, span_acc: 0.1328, em: 0.1392, f1: 0.2308, loss: 6.5829 ||: : 7095it [22:07,  5.35it/s]
start_acc: 0.2418, end_acc: 0.2631, span_acc: 0.1866, em: 0.1954, f1: 0.2801, loss: 6.2491 ||: : 888it [01:05, 13.51it/s]
start_acc: 0.2010, end_acc: 0.2101, span_acc: 0.1284, em: 0.1352, f1: 0.2267, loss: 6.6907 ||: : 7095it [22:05,  5.35it/s]
start_acc: 0.2489, end_acc: 0.2593, span_acc: 0.1808, em: 0.1863, f1: 0.2699, loss: 6.1341 ||: : 888it [01:06, 13.45it/s]
start_acc: 0.1732, end_acc: 0.1897, span_acc: 0.1085, em: 0.1138, f1: 0.2024, loss: 7.0391 ||: : 7095it [22:05,  5.35it/s]
start_acc: 0.1873, end_acc: 0.1993, span_acc: 0.1351, em: 0.1376, f1: 0.2170, loss: 6.7341 ||: : 888it [01:05, 13.61it/s]
start_acc: 0.0953, end_acc: 0.1067, span_acc: 0.0489, em: 0.0513, f1: 0.1171, loss: 8.3185 ||: : 7095it [22:04,  5.36it/s]
start_acc: 0.1017, end_acc: 0.1008, span_acc: 0.0380, em: 0.0386, f1: 0.1004, loss: 8.7326 ||: : 888it [01:06, 13.43it/s]
start_acc: 0.0488, end_acc: 0.0528, span_acc: 0.0181, em: 0.0194, f1: 0.0730, loss: 10.5009 ||: : 7095it [22:15,  5.31it/s]
start_acc: 0.0649, end_acc: 0.0503, span_acc: 0.0332, em: 0.0328, f1: 0.0723, loss: 9.4680 ||: : 888it [01:05, 13.58it/s]
"

with not lazy parameter:
"{'drop_out': 0.2, 'learning_rate': 0.001920946131952178, 'mini_batch': 64}
<class 'allennlp.data.tokenizers.character_tokenizer.CharacterTokenizer'>
56754it [02:38, 357.32it/s]
7100it [00:21, 334.84it/s]
63854it [02:11, 486.15it/s]
<class 'allennlp.modules.text_field_embedders.text_field_embedder.TextFieldEmbedder'>
100%|##########| 610977/610977 [00:30<00:00, 20281.57it/s]
<class 'allennlp.modules.seq2seq_encoders.seq2seq_encoder.Seq2SeqEncoder'>
/usr/local/lib/python3.6/dist-packages/torch/nn/modules/rnn.py:38: UserWarning: dropout option adds dropout after all but last recurrent layer, so non-zero dropout expects num_layers greater than 1, but got dropout=0.2 and num_layers=1
  "num_layers={}".format(dropout, num_layers))
<class 'allennlp.modules.similarity_functions.similarity_function.SimilarityFunction'>
<class 'allennlp.modules.seq2seq_encoders.seq2seq_encoder.Seq2SeqEncoder'>
<class 'allennlp.modules.seq2seq_encoders.seq2seq_encoder.Seq2SeqEncoder'>
start_acc: 0.2060, end_acc: 0.2243, span_acc: 0.1480, em: 0.1557, f1: 0.2408, loss: 6.3053 ||: 100%|##########| 887/887 [16:01<00:00,  1.02s/it]
start_acc: 0.3596, end_acc: 0.3746, span_acc: 0.2993, em: 0.3106, f1: 0.4081, loss: 4.7112 ||: 100%|##########| 111/111 [00:52<00:00,  1.45it/s]
start_acc: 0.4071, end_acc: 0.4376, span_acc: 0.3309, em: 0.3480, f1: 0.4743, loss: 4.4569 ||: 100%|##########| 887/887 [13:37<00:00,  1.21it/s]
start_acc: 0.4796, end_acc: 0.5011, span_acc: 0.4077, em: 0.4261, f1: 0.5459, loss: 3.8646 ||: 100%|##########| 111/111 [00:33<00:00,  1.96it/s]
start_acc: 0.5334, end_acc: 0.5663, span_acc: 0.4493, em: 0.4692, f1: 0.6081, loss: 3.4300 ||: 100%|##########| 887/887 [13:37<00:00,  1.12s/it]
start_acc: 0.5015, end_acc: 0.5238, span_acc: 0.4251, em: 0.4456, f1: 0.5696, loss: 3.7176 ||: 100%|##########| 111/111 [00:33<00:00,  1.96it/s]
start_acc: 0.6181, end_acc: 0.6435, span_acc: 0.5328, em: 0.5514, f1: 0.6851, loss: 2.7752 ||: 100%|##########| 887/887 [13:38<00:00,  1.21it/s]
start_acc: 0.5079, end_acc: 0.5279, span_acc: 0.4368, em: 0.4582, f1: 0.5774, loss: 3.7773 ||: 100%|##########| 111/111 [00:33<00:00,  1.98it/s]
start_acc: 0.6737, end_acc: 0.7020, span_acc: 0.5921, em: 0.6086, f1: 0.7370, loss: 2.3059 ||: 100%|##########| 887/887 [13:33<00:00,  1.07s/it]
start_acc: 0.5069, end_acc: 0.5370, span_acc: 0.4434, em: 0.4637, f1: 0.5800, loss: 3.9153 ||: 100%|##########| 111/111 [00:33<00:00,  1.99it/s]
start_acc: 0.7203, end_acc: 0.7429, span_acc: 0.6419, em: 0.6568, f1: 0.7753, loss: 1.9549 ||: 100%|##########| 887/887 [13:34<00:00,  1.11it/s]
start_acc: 0.5056, end_acc: 0.5308, span_acc: 0.4341, em: 0.4549, f1: 0.5762, loss: 4.1654 ||: 100%|##########| 111/111 [00:33<00:00,  1.96it/s]
start_acc: 0.7518, end_acc: 0.7747, span_acc: 0.6788, em: 0.6921, f1: 0.8030, loss: 1.6881 ||: 100%|##########| 887/887 [13:35<00:00,  1.31it/s]
start_acc: 0.5110, end_acc: 0.5228, span_acc: 0.4348, em: 0.4589, f1: 0.5786, loss: 4.2740 ||: 100%|##########| 111/111 [00:34<00:00,  1.98it/s]
start_acc: 0.7777, end_acc: 0.7986, span_acc: 0.7090, em: 0.7186, f1: 0.8233, loss: 1.4938 ||: 100%|##########| 887/887 [13:35<00:00,  1.14it/s]
start_acc: 0.4983, end_acc: 0.5231, span_acc: 0.4296, em: 0.4501, f1: 0.5686, loss: 4.4683 ||: 100%|##########| 111/111 [00:33<00:00,  2.01it/s]
start_acc: 0.7996, end_acc: 0.8177, span_acc: 0.7328, em: 0.7408, f1: 0.8389, loss: 1.3390 ||: 100%|##########| 887/887 [13:35<00:00,  1.29it/s]
start_acc: 0.4999, end_acc: 0.5204, span_acc: 0.4273, em: 0.4479, f1: 0.5679, loss: 4.7711 ||: 100%|##########| 111/111 [00:33<00:00,  2.00it/s]
start_acc: 0.8308, end_acc: 0.8471, span_acc: 0.7684, em: 0.7755, f1: 0.8651, loss: 1.0815 ||:  26%|##6       | 231/887 [03:38<09:28,  1.15it/s]
"
Any idea why there is a huge difference between EM in the corresponding epochs?

**Comment by Senagb** [comment]
Sorry false alarm, I have figured out the issue it was in my dataset.

**Comment by Senagb** [comment]
@DeNeutoy  Thanks it seems it is like you say as the problem in my dataset wasn't the problem.

**Comment by susht3** [comment]
i have the same problem, when i use lazy my f1 is very small
